openedx/features/course_experience/api/v1/views.py

```diff
@@ -30,3 +36,44 @@ def reset_course_deadlines(request):
         return Response({'message': 'Deadlines successfully reset.'})
     except Exception:
         raise UnableToResetDeadlines
+
+
+class CourseDeadlinesMobileView(RetrieveAPIView):
+    """
+    **Use Cases**
+
+        Request course deadline info for mobile
+
+    **Example Requests**
+
+        GET api/course_experience/v1/course_deadlines_info/{course_key}
+
+    **Response Values**
+
+        Body consists of the following fields:
+
+        missed_deadlines: (bool) Whether the user has missed any graded content deadlines for the given course.
+        missed_gated_content: (bool) Whether the user has missed any gated content for the given course.
+        content_type_gating_enabled: (bool) Whether content type gating is enabled for this enrollment.
+        verified_upgrade_link: (str) The URL to ecommerce IDA for purchasing the verified upgrade.
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 401 if the user is not authenticated.
+        * 404 if the course is not available or cannot be seen.
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (IsAuthenticated,)
+    serializer_class = CourseDeadlinesMobileSerializer
+
+    def get(self, request, *args, **kwargs):
+        context = self.get_serializer_context()
+        serializer = self.get_serializer_class()(context=context)
```

Comment:
I think you can just call `self.get_serializer()`

Reply:
Do we even need to overwrite the get method here?  i forget how drf is set up but doesnt it just work out of the box if you dont want any custom functionality